bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -1332,7 +1332,7 @@ private void checkStatus() {
                                       !statusMessage.contains(TEMPORARY_ISSUE_FAILURE_MARKER);
             }
 
-            if (approvalNeeded()) {
+            if (approvalNeeded() && visitor.messages().isEmpty()) {
```

Comment:
Is this enough? Shouldn't we check `readyForReview` as well. What about the other things in the list for `readyForIntegration`?

Reply:
You are right. I will fix the problem soon. Thank you.

Reply:
I also refactored the code about `readyForIntegration`.